Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -330,6 +335,21 @@ public virtual void OnApplicationQuit()
         /// </summary>
         public virtual void OnValidate()
         {
+            // add authenticator if there is none yet. makes upgrading easier.
+            if (authenticator == null)
+            {
+                // was an authenticator added yet? if not, add one
+                authenticator = GetComponent<NetworkAuthenticator>();
+                if (authenticator == null)
+                {
+                    authenticator = gameObject.AddComponent<NetworkAuthenticator>();
+                    Debug.Log("NetworkManager: added missing NetworkAuthenticator.");
+                }
+#if UNITY_EDITOR
+                UnityEditor.EditorUtility.SetDirty(gameObject);
```

Comment:
```suggestion
                Undo.RecordObject(this, "Added Authenticator");
```

SetDirty was the old way to do it. Undo.RecordObject is what works properly now.